### PR TITLE
[StateAccumulator] Exercise transaction edge cases in test

### PR DESCRIPTION
**Pull request by williampsmith**
## Description 

This exercises state accumulation against edge case transactions such as those created from the use of `created_then_wrapped`, `unwrapped_then_deleted`, as well as other combinations of wraps and unwraps, by adding incremental checking of the live object set against the result of accumulating single transaction effects. This path will be exercised by any tests that hit `authority_tests::send_and_confirm_transaction_()`. Notably, this includes all tests in `move_interation_tests.rs`. 

Included also is bug fixes uncovered from the above.

## Test Plan 

```
cargo simtest && cargo nextest run
```

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-core/src/state_accumulator.rs [hunk 1]
```diff
@@ -98,45 +115,88 @@ impl StateAccumulator {
...
-
         acc.remove_all(modified_at_digests);
 
-        epoch_store.insert_state_hash_for_checkpoint(&checkpoint_seq_num, &acc)?;
-        debug!("Accumulated checkpoint {}", checkpoint_seq_num);
+        // Process unwrapped and unwrapped_then_deleted effects, which need to be
+        // removed as WrappedObject using the last sequence number it was tombstoned
+        // against. Unfortunately since this happened in a past transaction, and the
+        // child object may have been modified since (and hence its sequence number incremented),
+        // we must traverse the objects table until we find the tombstone in order to get the
```
Comment (mystenmark) [inline crates/sui-core/src/state_accumulator.rs:171]: this comment is a bit out of date now - we don't really traverse, we can seek right to it

**Review by mystenmark — APPROVED** [review]
